### ISSUE TITLE
Implement support for table functions

### ISSUE THREAD

**Issue by beikov** [Blazebit/blaze-persistence#181]
There are some cases that we need to cover
* unnest from an array column like ```select unnest(table.arrayColumn)```
* table generating functions like ```select * from generate_series(1,10)```
* plain values like ```select * from (values (1, 'abc'), (2, 'def'))```

**Comment by beikov** [comment]
The only way to support that in my opinion is to register some virtual entities like done with CTEs and replace them in the generated SQL with the vendor specific syntax.

**Comment by beikov** [comment]
This might also be a nice function that we could try to integrate at least in a test to see if entity functions can be applied in that case.

**Comment by beikov** [comment]
I just had a thought that since entity array expressions are a thing now, it would be nice if we could use `VALUES` or `generate_series` like an entity array expression. I'm thinking about using a secondary entity view root like e.g. `@EntityViewRoot(name = "days", expression = "generate_series(CURRENT_DATE, :param, interval('1 day'))", condition = "...")` to generate a few days for e.g. grouping/aggregating. Not sure yet what kind of syntax we could introduce for actual values.

**Comment by jwgmeligmeyling** [comment]
`select unnest(table.arrayColumn)` already works, I suppose you mean `select * from unnest(table.arrayColumn)`?

How would we go about implementing the simple cases of `unnest` and `generate_series` with the existing `ENTITY_FUNCTION`? I believe its capabilities improved quite significantly since the inline subquery rendering / lateral join support?

**Comment by beikov** [comment]
>select unnest(table.arrayColumn) already works, I suppose you mean select * from unnest(table.arrayColumn)?

Yes, exactly. Only PG is so lenient to allow using `unnest` in a scalar expression context. Also, other table functions require the use in the `FROM` clause.

>How would we go about implementing the simple cases of unnest and generate_series with the existing ENTITY_FUNCTION? I believe its capabilities improved quite significantly since the inline subquery rendering / lateral join support?

Yes, I tried to model the `ENTITY_FUNCTION` infrastructure in a generic fashion to eventually allow users to provide custom functions, or at least make it pluggable for our own purposes, so ideally we would extract some sort of SPI out of this to allow implementing other table functions.
